muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java

```diff
@@ -887,6 +887,10 @@ public synchronized String generateUniqueUrlName(WorkspaceNode parent, String ti
 
   public synchronized String generateUniqueUrlName(WorkspaceNode parent, WorkspaceNode targetNode, String title) {
     String urlName = generateUrlName(title);
+    String fileName = StringUtils.substringBeforeLast(urlName, ".");
```

Comment:
FilenameUtils.getBaseName is only suitable for dealing with file names. This code deals with titles (and occasional file names). In other words, if this method was given a string "1. luku", using FilenameUtils would turn it into name "1" and extension " luku", leading to unpredictable behaviour further down the line.